### DISCUSSION TITLE
Redaction not applied when save to png

### DISCUSSION THREAD

**Discussion by lunamaze** [pymupdf/PyMuPDF#3486] [Looking for help]
I have a page of pdf file with an unwanted image. I used `page.apply_redactions(graphics=0)` and the image is visually disappeared (I think it is still in the pdf source file, but just covered with a white/transparent rectangle?)

After that I use `get_pixmap` and `save` to save the page into png file, yet the image re-appeared. Is there any way that I can save the pdf file to image while applying the redactions?

**Comment by JorjMcKie** [comment]
You seem to confuse the terms "image" and "graphics".
**_Image_** refers to embedded images, while **_graphics_** are vector graphics. Two completely different things.
To remove an image, you must specify `PDF_REDACT_IMAGE_REMOVE` when applying redactions: `page.apply_redactions(images=fitz.PDF_REDACT_IMAGE_REMOVE)`.
